options.go

```diff
@@ -106,14 +106,33 @@ func WithHostPortAccess(ports ...int) CustomizeRequestOption {
 	}
 }
 
+// WithName will set the name of the container.
+func WithName(containerName string) CustomizeRequestOption {
+	return func(req *GenericContainerRequest) error {
+		if containerName == "" {
+			return errors.New("container name must be provided")
+		}
```

Comment:
question: is this check needed? I ask as this may be used to unset a name, which this prevents. If it's kept we should document it in the doc comment.

If removed we'd need to add that to WithReuseByName

Reply:
Containers must come without names, by default, so users would just want to add a name, not cleaning up an existing one.

This related to the discussion we had about booleans: if at some point we have all fields private, only accessible via functional opts, then we can deprecate the GenericContainerRequest struct and "clone" it into an internal/private struct that will hold the state for the user request. Then, the opinionated semantics for not accepting true/false (for booleans) or changing the container name (here) would enforce the expressiveness of the API